pkg/util/resourceinformer/resourceinformer.go

```diff
@@ -0,0 +1,155 @@
+package resourceinformer
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/keys"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+	"github.com/karmada-io/karmada/pkg/util/restmapper"
+)
+
+var _ ResourceInformer = &resourceInformerImpl{}
+
+type ResourceInformer interface {
```

Comment:
I like the idea of abstracting the informer-related functionalities.
It is another informer manager based on [genericmanager.MultiClusterInformerManager](https://github.com/karmada-io/karmada/tree/master/pkg/util/fedinformer/genericmanager) with more context of Karmada resource types(such as it recognizes `Cluster`, `Work`).

 I just have a feeling that the name of it (`ResourceInformer`) seems `a little bit generic` and doesn't show its responsibility.
I'm not sure if `MemberClusterInformer` would be better. Looking forward to your opinions.

Reply:
I just keep the name as what it was in `workstatus controller`.  
I think it is ok to use `MemberClusterInformer` to instead. I will fix it soon.

Reply:
Done.